docs/sources.md

```diff
@@ -0,0 +1,246 @@
+# Sources
+
+A "source" in Dalec is an abstraction for fetching dependencies of a build.
+Usually this is source code but technically it could be anything.
+The source abstraction enables fetching build sources over various protocols.
+
+Some sources are considered as inherently file-based sources, like HTTP URLs or local directories.
+Other sources are considered as inherently directory-based sources, like git repositories.
+Depending on the source type, the behavior of certain things may be different, depending on the target implementation (e.g. mariner2, jammy, windows, etc).
+Sources are injected into the root path of the build environment using the name of the source.
+
+Ideally the content of a source is platform agnostic (e.g. no platform specific binaries).
+These sources are used to create source packages for the target platform, such as an srpm or a debian dsc.
+
+## Top-level source configuration
+
+For all source types, you can specify the following top-level configuration:
+
+- `path`: The path to extract from the source type
+- `includes`: A list of glob patterns to include from the source
+- `excludes`: A list of glob patterns to exclude from the source
```

Comment:
Could we maybe have some examples using these options or adding to existent examples?